Sources/SwiftGenKit/Parsers/Strings/StringsParser.swift

```diff
@@ -9,21 +9,42 @@ import PathKit
 
 public enum Strings {
   public enum ParserError: Error, CustomStringConvertible {
-    case duplicateTable(name: String)
-    case failureOnLoading(path: String)
-    case invalidFormat
+    case duplicateTableFile(path: Path, existing: Path)
+    case failureOnLoading(path: Path)
+    case invalidFormat(reason: String)
     case invalidPlaceholder(previous: Strings.PlaceholderType, new: Strings.PlaceholderType)
+    case invalidPluralFormat(missingVariableKey: String, pluralKey: String)
+    case invalidVariableRuleValueType(variableName: String, valueType: String)
+    case unsupportedFileType(path: Path, supported: [String])
 
     public var description: String {
       switch self {
-      case .duplicateTable(let name):
-        return "Table \"\(name)\" already loaded, cannot add it again"
+      case .duplicateTableFile(let path, let existing):
+        return """
+        Duplicate file of the same type for table "\(path.lastComponentWithoutExtension)":
+        Trying to load \(path) while we've already loaded \(existing)
+        """
       case .failureOnLoading(let path):
         return "Failed to load a file at \"\(path)\""
-      case .invalidFormat:
-        return "Invalid strings file"
+      case .invalidFormat(let reason):
+        return "Invalid format. \(reason)"
       case .invalidPlaceholder(let previous, let new):
         return "Invalid placeholder type \(new) (previous: \(previous))"
+      case .invalidPluralFormat(let missingVariableKey, let pluralKey):
+        return """
+        The variable \"\(missingVariableKey)\" referenced in the NSStringLocalizedFormatKey is not \
+        defined in the variables of the plural key \"\(pluralKey)\"
+        """
+      case .invalidVariableRuleValueType(let variableName, let valueType):
+        return """
+        The variable \"\(variableName)\" has sepcified \"\(valueType)\" as its NSStringFormatValueTypeKey. \
```

Comment:
Small typo.

```suggestion
        The variable \"\(variableName)\" has specified \"\(valueType)\" as its NSStringFormatValueTypeKey. \
```

Just tried out this branch and got this error. We have `.stringsdict`s that use  `@` as `NSStringFormatValueTypeKey`. We’d love to see support for it in a future version 👍

Reply:
Thanks @thii for this feedback. I'll fix the typo.

Regarding the use case / error you mention, this is an invalid usage of `stringsdict`, as least according to Apple's StringDict documentation, as well as according to the various tests we did with Foundation in playgrounds or example projects.

This actually has nothing to do with SwiftGen, but with how Foundation works: the `NSStringFormatValueTypeKey` can only be a placeholder specifier for a number (`d`, `f`, …) and can't be `@` because the parameter it corresponds to is used by `Foundation` at runtime to determine the plurality and thus determine what plural form to use. When you think about it, it actually makes sense because how would you expect `Foundation` to know which of the `zero`, `one` or `other` plural form if what you pass it is `String(format: "Hello %@", name)`? `name` being a String / object and not a number, there would be no way for Foundation to determine at runtime if name was supposed to be singular or plural and determine which plural form of the translation to use.

Reply:
I see. It’s probably one of our hacks to add plurals support to SwiftGen. I’ll take a look further.